File: VB/GridMVVMBandsSample/BandTemplateSelector.vb
Imports System.Windows
Imports System.Windows.Controls

Namespace GridMVVMBandsSample
	Public Class BandTemplateSelector
		Inherits DataTemplateSelector

		Public Property SingleColumnBandTemplate() As DataTemplate

		Public Property MultiColumnBandTemplate() As DataTemplate

		Public Overrides Function SelectTemplate(ByVal item As Object, ByVal container As DependencyObject) As DataTemplate
			Dim band As Band = TryCast(item, Band)
			If band Is Nothing Then
				Return Nothing
			End If
			If band.ChildColumns.Count = 1 Then
				Return SingleColumnBandTemplate
			End If
			Return MultiColumnBandTemplate
		End Function
	End Class
End Namespace

File: CS/GridMVVMBandsSample/BandTemplateSelector.cs
using System.Windows;
using System.Windows.Controls;

namespace GridMVVMBandsSample {
    public class BandTemplateSelector : DataTemplateSelector {

        public DataTemplate SingleColumnBandTemplate { get; set; }
        public DataTemplate MultiColumnBandTemplate { get; set; }

        public override DataTemplate SelectTemplate(object item, DependencyObject container) {
            Band band = item as Band;
            if(band == null) return null;
            if(band.Header == "Position") {
                return SingleColumnBandTemplate;
            }
            return MultiColumnBandTemplate;
        }
    }
}
